working/augmentation-libraries/feature-specification.md

```diff
@@ -724,11 +793,11 @@ It is a **compile-time error** if:
 
 *   An `abstract` variable is augmented with a non-abstract variable.
 
-*   An `external` variable is augmented with an `abstract` variable.
+*   An `external` variable is augmented with a variable declaration.
```

Comment:
I am not sure if we need this restriction? It is true that the `external` keyword would be unnecessary though in this case, but I feel like we have gone back and forth on this a bit so and I am curious what the motivation for not allowing this would be?

Do we need to make similar changes for external functions? The CFE today actually requires us to mark certain things external (but this is a bug).

Reply:
> I think it was mainly that an external "variable declaration" doesn't introduce a variable to augment, it only introduces a getter and a setter. The only augmentations you should be allowed to make are to getter and setter.

That is a compelling argument for me. I think we can leave things as you have defined here. Possibly expand to allow adding metadata/comments.

> ```dart
> @newData
> augment external int foo; // Reasonable, but looks like the augmentation itself is external.
> ```

This seems ok to me, it just adds the annotation.

> ```dart
> @newData
> augment int foo; // Possible, looks like it's augmenting a variable.
> ```

I don't like this, because it does seem like it is actually making it non-external. And, we don't allow augmenting getters/setters with variables.

> ```dart
> @newData
> augment abstract int foo; // Already not allowed (`abstract` and `external` are usually mutually exclusive).
> ```

lgtm

> ```dart
> augment int foo = 1; // Completely unacceptable.
> ```

agreed (now)

> ```dart
> @newData
> augment foo; // Nice-to-have, but not currently possible.
> ```

I think this looks a bit weird so I wouldn't try to allow it